client/internal/receiver.go

```diff
@@ -168,3 +169,14 @@ func (r *Receiver) processErrorResponse(body *protobufs.ServerErrorResponse) {
 func (r *Receiver) rcvAddonsAvailable(addons *protobufs.AddonsAvailable) {
 	// TODO: implement this.
 }
+
+func (r *Receiver) rcvAgentIdentification(ctx context.Context, agentId *protobufs.AgentIdentification) {
+	if agentId == nil {
+		return
+	}
+
+	err := r.callbacks.OnAgentIdentification(ctx, agentId)
```

Comment:
I think in addition to calling the callback we need to update the instanceUid in the Sender (if the callback doesn't return an error).